indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -746,10 +777,16 @@ private void handleStatus(final TaskStatus status)
                  .addData("statusCode", status.getStatusCode())
                  .emit();
             }
+            finally {
+              statusUpdatesInQueue.decrementAndGet();
+              handledStatusUpdates.incrementAndGet();
+            }
           }
-        }
+        },
+        // The onSuccess and onFailure should be lightweight enough
+        // so that the executor giving the statusFuture is not blocked
```

Comment:
```suggestion
        // The onSuccess and onFailure should be lightweight enough
        // so that the executor giving the statusFuture is not blocked
        // We use the direct executor here so we can track in-flight status updates. 
        // so the metrics for in-flight status gets updated immediately. 
```

Reply:
Simplified this comment, added the rest of the info in the javadoc of this method as it seemed like a better place for detailed richer content.